File: comu/com.energyict.mdc.firmware/src/main/java/com/energyict/mdc/firmware/DeviceFirmwareHistory.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.firmware;

import java.util.List;

public interface DeviceFirmwareHistory {

    /**
     * The list of {@link DeviceFirmwareVersionHistoryRecord}s for given device
     * @return the list of {@link ActivatedFirmwareVersion} owned by given device
     */
    List<DeviceFirmwareVersionHistoryRecord> history();

    /**
     * The list of {@link DeviceFirmwareVersionHistoryRecord}s for given
     * firmwareType owned by given device
     * @param firmwareType the firmware type to match
     * @return the list of {@link ActivatedFirmwareVersion} owned by given device
     */
    List<DeviceFirmwareVersionHistoryRecord> history(FirmwareType firmwareType);



}

File: comu/com.energyict.comserver.comserver.core.extjs/src/main/web/js/mdc/src/model/DeviceFirmwareHistory.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Mdc.model.DeviceFirmwareHistory', {
    extend: 'Ext.data.Model',
    fields: [
        'firmwareVersion',
        'firmwareType',
        'activationDate'
    ]
});
